src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -630,7 +589,7 @@ public void WriteStartObject()
         private void WriteStart(byte token)
         {
             if (CurrentDepth >= _options.MaxDepth)
-                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.DepthTooLarge, _currentDepth, _options.MaxDepth, token: default, tokenType: default);
+                ThrowInvalidOperationException_DepthTooLarge();
```

Comment:
This predates your own changes, but it looks like this method is mixing branching styles. Could you change this so that it's also a block?